projects/telemetry/inc/gps.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+// This will be the GPS driver. It will be responsive for initializing, and cleaning up the GPS
+// as well as providing data from the GPS.
+
+// On GPS data: the GPS sends NMEA (National Marine Electronics Association) messages which
+// contain data about current time, position, speed, etc. The message we are most interested
+// in is the GGA message (Global Positioning System Fix Data). It contains latitude and
+// longitude.
+#include <stdbool.h>
+#include "status.h"
+#include "uart.h"
+
+// Just some constants so that the max length of raw data can be set.
+// A GGA message will be around a hundred characters.
+#define GPS_MAX_NMEA_LENGTH 128
+
+// This struct basically contains all the info about pins etc.
+// Check this document on page 4:
+// https://www.linxtechnologies.com/wp/wp-content/uploads/evm-gps-f4.pdf
+typedef struct {
+  UARTPort port;
+  UARTSettings *uart_settings;    // The uart settings to be used
+  GPIOSettings *settings_power;   // Pin which provides power
```

Comment:
Do these `GPIOSettings` really need to be exposed? You're setting both of these to `telemetry_settings_gpio_general` anyways.

I think you can just statically allocate this in your `init` function, and then use the same `GPIOSettings` for everything. You're assigning everything to `telemetry_settings_gpio_general` anyways.

```c
GPIOSettings telemetry_settings_gpio_general = {
  .direction = GPIO_DIR_OUT,  // The pin needs to output.
  .state = GPIO_STATE_LOW,    // Start in the "off" state.
  .alt_function = GPIO_ALTFN_NONE,
};
```